zrml/swaps/src/lib.rs

```diff
@@ -1308,10 +1311,19 @@ mod pallet {
 
             <Pools<T>>::insert(next_pool_id, Some(pool.clone()));
 
-            Self::deposit_event(Event::PoolCreate(
-                CommonPoolEventParams { pool_id: next_pool_id, who },
-                pool,
-            ));
+            if scoring_rule == ScoringRule::CPMM {
+                Self::deposit_event(Event::PoolCreate(
+                    CommonPoolEventParams { pool_id: next_pool_id, who },
+                    pool,
+                    amount,
+                ));
+            } else {
+                Self::deposit_event(Event::PoolCreate(
+                    CommonPoolEventParams { pool_id: next_pool_id, who },
+                    pool,
+                    0u32.saturated_into(),
+                ));
+            }
```

Comment:
We could use a ternary operator here to improve style. I am fine to keep it like that, that PR already demanded enough effort from us.

```suggestion
            Self::deposit_event(Event::PoolCreate(
                CommonPoolEventParams { pool_id: next_pool_id, who },
                pool,
                if scoring_rule == ScoringRule::CPMM { amount } else { 0u32 },
            ));
```